File: Dualroots/Dualroots/ClientApp/src/app/models/design.ts
export class Design{
    id:any
    commentId:any
    index:any
    subject:string
    comment:string
    commentOn:string
    file:string
    class:string
    type:string
    isSelectedCol:boolean
    isActive:boolean
    isEditable:boolean
    editIndex:boolean
    editCommentId:boolean
    editSubject:boolean
    editComment:boolean
    editDate:boolean
    editFile:boolean
    editCommentOn:boolean
    editClass:boolean
    editType:boolean
    editableField:string
    date:Date
    isMandatory:boolean
    mandatroyColumns:string[]
    editableColumns:string[]=[]
    constructor(args:any={}){
        this.id= args.id
        this.commentId = args.commentId
        this.index = args.index
        this.subject = args.subject
        this.comment = args.comment
        this.commentOn = args.commentOn
        this.file = args.file
        this.class = args.class
        this.type = args.type
        this.isEditable = false
        this.editableField= args.editableField
        this.editCommentId=false
        this.editIndex=false
        this.editSubject=false
        this.editClass=false
        this.editCommentOn=false
        this.editDate=false
        this.editComment=false
        this.editType=false
        this.date = args.date
        this.isMandatory=Boolean(args.isMandatory)
        this.mandatroyColumns = args.mandatroyColumns==null?[]:JSON.parse(args.mandatroyColumns) 
        this.editableColumns = args.editableColumns==null? []: JSON.parse(args.editableColumns)
        this.isSelectedCol=false
        this.isActive=false
    }
}

File: Dualroots/Dualroots/Models/Design.cs
using System;
using System.Collections.Generic;

namespace Dualroots.Models;

public partial class Design
{
    public int Id { get; set; }

    public int? CommentId { get; set; }

    public string? Index { get; set; }

    public string? Subject { get; set; }

    public string Comment { get; set; } = null!;

    public string CommentOn { get; set; } = null!;

    public string File { get; set; } = null!;

    public string Class { get; set; } = null!;

    public string Type { get; set; } = null!;

    public string? EditableField { get; set; }

    public DateTime? Date { get; set; }

    public int? IsMandatory { get; set; }

    public string? EditableColumns { get; set; }

    public string? MandatroyColumns { get; set; }
}
